server/src/main/java/keywhiz/service/daos/SecretController.java

```diff
@@ -179,9 +197,11 @@ public SecretBuilder withGenerationOptions(Map<String, String> generationOptions
      * @return an instance of the newly created secret.
      */
     public Secret build() {
-      try {
         secretDAO.createSecret(name, encryptedSecret, version, creator, metadata, description, type, generationOptions);
         return transformer.transform(secretDAO.getSecretByNameAndVersion(name, version).get());
+/*
+        // TODO: I don't know if propagating the DataAccessException is the right thing to do.
```

Comment:
Two questions:
1. should we catch DataAccessException (which is a jooq thing) and throw something more generic?
2. does DataAccessException leak any information we don't want to leak (sensitive data from raw SQL)?